apps/src/schoolInfo/hooks/useSchoolInfo.tsx

```diff
@@ -68,18 +68,60 @@ export function useSchoolInfo(initialState: SchoolInfoInitialState) {
     [initialState.schoolName]
   );
 
+  const [state, setState] = useState<{
+    country: string;
+    schoolId: string;
+    schoolZip: string;
+    schoolName: string;
+    schoolsList: SchoolDropdownOption[];
+  }>({
+    country: detectedCountry,
+    schoolId: detectedSchoolId,
+    schoolZip: detectedZip,
+    schoolName: detectedSchoolName,
+    schoolsList: [],
+  });
+
   // State hooks
-  const [country, setCountry] = useState(detectedCountry);
-  const [schoolId, setSchoolId] = useState(detectedSchoolId);
-  const [schoolZip, setSchoolZip] = useState(detectedZip);
-  const [schoolName, setSchoolName] = useState(detectedSchoolName);
-  const [schoolsList, setSchoolsList] = useState<SchoolDropdownOption[]>([]);
+  const setCountry = (value: string) => {
```

Comment:
It seems like a lot of additional setState code to empower a reset to only have to call setState once. But that's a nit and of course just a style preference!

Reply:
definitely more boilerplate this way to be sure!